File: Offtopics/Sipahi_float_problem/precisao.f90
program checkprecision
   implicit none
   real :: teste, check, step= 2.0
   integer :: i

   teste = 1.0
   do
      check = 1.0 + teste
      if(check == 1.0) exit
      teste = teste / step
      !print *, 'teste is now', teste 
   end do

   print *, 'precision is', teste 

   teste = 1.0
   do
      !check = 1.0 + teste
      if((1.0 + teste) == 1.0) exit
      teste = teste / step
      !print *, 'teste is now', teste 
   end do

   print *, 'precision is', teste 

   teste = 1.0
   do while( (1.0 + teste) /= 1.0 )
      teste = teste / step
      !print *, 'teste is now', teste 
   end do

   print *, 'precision is', teste 

   i = 0
   do while( (1.0 + step**(-i)) /= 1.0 )
      i=i+1
      !print *, 'teste is now', teste 
   end do

   print *, 'precision is', 10.0**(-i)

end program checkprecision

File: Offtopics/Sipahi_float_problem/precisao.c
main(){
   printf("Size of float %u", sizeof(float));
   float test;
   float step;
   test = 1.0;
   step = 2.0;
   for(int i=0;i<100;i++){
      if((1.0 + test) == 1.0) break;
      test = test / step;
      //printf("%g\n",test);
   }
   printf("%g\n",test);
}
